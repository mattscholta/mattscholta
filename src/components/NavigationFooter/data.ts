import { LinkProps } from 'next/link';

export interface Data extends LinkProps {
  copy: string;
}

export const data: Data[] = [
  {
    copy: 'Home',
    href: '/'
  },
  {
    copy: 'About',
    href: '/about'
  },
  {
    copy: 'Contact',
    href: '/contact'
  }
];
