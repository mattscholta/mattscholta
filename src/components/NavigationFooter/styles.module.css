.component {
  background-color: theme('colors.gray.0');
  border: 1px solid theme('colors.gray.3');
  color: theme('colors.secondary');
  font-family: theme('fontFamily.heading');
  font-size: theme('fontSize.14');
  height: var(--footer-height);
  padding: theme('spacing.2x');
}

.links {
  display: flex;
  gap: theme('spacing.2x');
  justify-content: center;
}

.link {
  color: theme('colors.secondary');
  padding: theme('spacing.1x');
  text-decoration: none;
}

.link.active {
  color: theme('colors.primary');
}
