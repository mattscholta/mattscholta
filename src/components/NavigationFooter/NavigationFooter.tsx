import * as React from 'react';
import classnames from 'classnames';
import Link from 'next/link';

import { Routes, useRoutes } from '@src/hooks/useRoutes';
import { Data, data } from './data';

import styles from './styles.module.css';

interface NavigationFooterProps {
  className?: string;
}

/**
 * @name NavigationFooter
 * @description tbd...
 */
const NavigationFooter: React.FC<NavigationFooterProps> = (props) => {
  const { className } = props;

  // Hooks
  const { active } = useRoutes();

  // Setup
  const date = new Date();
  const year = date.getFullYear();

  // Styles
  const cssComponent = classnames(styles.component, className);

  // Markup
  const renderLink = (link: Data) => {
    const key = link.copy.toLowerCase();
    const cssLink = classnames(styles.link, {
      [styles.active]: active[key as Routes]
    });

    return (
      <Link {...link} key={link.copy}>
        <a className={cssLink}>{link.copy}</a>
      </Link>
    );
  };

  return (
    <div className={cssComponent}>
      <div className="ui-container-xl u-text-center">
        <div className={styles.links}>{data.map(renderLink)}</div>
        <p>
          &copy; {year} - Built with{' '}
          <span className="u-text-primary">&hearts;</span> in San Diego, CA.
        </p>
      </div>
    </div>
  );
};

export { NavigationFooter };
export type { NavigationFooterProps };
