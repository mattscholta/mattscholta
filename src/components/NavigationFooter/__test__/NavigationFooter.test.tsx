import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { NavigationFooter, NavigationFooterProps } from '../NavigationFooter';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '/',
      push: jest.fn(),
      query: {},
      route: '/'
    };
  }
}));

describe('NavigationFooter', () => {
  let props: NavigationFooterProps;
  let wrapper: RenderResult;

  const createWrapper = () => <NavigationFooter {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
