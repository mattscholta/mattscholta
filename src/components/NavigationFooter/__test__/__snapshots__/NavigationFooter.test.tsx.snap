// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationFooter matches snapshot 1`] = `
NodeList [
  <div
    class="component"
  >
    <div
      class="ui-container-xl u-text-center"
    >
      <div
        class="links"
      >
        <a
          class="link"
          href="/"
        >
          Home
        </a>
        <a
          class="link"
          href="/about"
        >
          About
        </a>
        <a
          class="link"
          href="/contact"
        >
          Contact
        </a>
      </div>
      <p>
        © 
        2021
         - Built with
         
        <span
          class="u-text-primary"
        >
          ♥
        </span>
         in San Diego, CA.
      </p>
    </div>
  </div>,
]
`;
